taxi/tests/test_views.py

```diff
@@ -0,0 +1,138 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_URL = reverse("taxi:car-list")
+DRIVER_URL = reverse("taxi:driver-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_public_manufacturer(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="Testusername", password="testpass123456"
+        )
+        self.client.force_login(self.user)
+
+    def test_private_manufacturer(self):
+        Manufacturer.objects.create(name="Test Manufacturer")
+        Manufacturer.objects.create(name="Test Manufacturer 2")
+
+        response = self.client.get(MANUFACTURER_URL)
+
+        manufacturer = Manufacturer.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturer)
+        )
+        self.assertTemplateUsed(response.context)
+
+    def test_manufacturer_search(self):
+        Manufacturer.objects.create(name="Ford", country="United States")
+        response = self.client.get(MANUFACTURER_URL + "?name=ford")
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")

```

Comment:
did u test search feature here?
same for others

Reply:
do you mean that it is better to do them in test_forms or what?

Reply:
i mean u are checking statuscode + template used, are u checked the context of response ? If search feature doesn't work and page will return all cars from db, will this test check this ? can u check search feature by creating only one car for this test ? use setUp for obj creating for all tests 
same for other tests
also add toggle assing me to car test + admin page tests